src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Controller/EntityControllerTrait.php

```diff
@@ -140,6 +140,12 @@ private function handleImport(Request $request, SaveSamlEntityCommand $command)
      */
     private function publishEntity(Entity $entity, FlashBagInterface $flashBag)
     {
+        if ($entity->getProtocol() === Entity::TYPE_OPENID_CONNECT) {
```

Comment:
The checks in the controller could then also use the `isReadOnly` method from the entity and doesn't have to do domain related stuff.